beancount_import/source/schwab_csv.py

```diff
@@ -219,6 +224,9 @@ class RawBrokerageEntry(RawEntry):
     quantity: Optional[Decimal]
     price: Optional[Decimal]
     fees: Optional[Decimal]
+    merger_symbol: Optional[str]
+    merger_quantity: Optional[Decimal]
+    merger_description: Optional[str]
```

Comment:
This isn't ideal as you note.
It's beyond the scope of this PR, but given Schwab's extremely limited CSV format requiring them to do ugly things like break a single transaction into multiple rows it is better to have flexibility when parsing. If we move away from the thin simple parser that just dumps CSV rows into `RawBrokerageEntry` and instead move the logic of `get_processed_entry` into the CSV reading logic it would be a net improvement (in this programmer's opinion). If we had that structure you'd construct the Merger data class in the CSV parsing loop.

The approach you take here isn't terrible though and I would be fine with it if it were my codebase. If we ended up finding several more transaction types like this then it will get really ugly, but I think it's unlikely there are too many transaction types that span multiple entries like this.

Reply:
I think we should defer the act moving the logic you mention into the CSV reading logic until another such situation arises. It will at least be more clear how to do it then.

Reply:
Makes sense to me.

Reply:
I'm not sure it really helps to move more logic into CSV parsing. Either way we end up having to do the same thing to handle a case like this (loop over sequential rows with some aggregation logic). The intention of the design is that the `Raw*` objects should be a very close mirror of exactly what is in the CSV file, and then in a separate step we process those raw entries into something more useful for importing into beancount. This separates the CSV-parsing details from the logical interpretation of the CSV rows actually mean. To me this seems like useful separation of concerns, and I don't see what is gained by collapsing them into a single step.